src/components/assistant-code-block.tsx

```diff
@@ -0,0 +1,121 @@
+import { Spinner } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useEffect } from 'react';
+import { ExtraProps } from 'react-markdown';
+import stripAnsi from 'strip-ansi';
+import { useExecuteWPCLI } from '../hooks/use-execute-cli';
+import Button from './button';
+import { ChatMessageProps } from './chat-message';
+import { CopyTextButton } from './copy-text-button';
+import { ExecuteIcon } from './icons/execute';
+
+type ContextProps = Pick<
+	ChatMessageProps,
+	'blocks' | 'updateMessage' | 'projectPath' | 'messageId'
+>;
+
+type CodeBlockProps = JSX.IntrinsicElements[ 'code' ] & ExtraProps;
+
+export default function createCodeComponent( contextProps: ContextProps ) {
+	return ( props: CodeBlockProps ) => <CodeBlock { ...contextProps } { ...props } />;
+}
+
+function CodeBlock( props: ContextProps & CodeBlockProps ) {
+	const content = String( props.children ).trim();
+	const containsWPCommand = /\bwp\s/.test( content );
+	const wpCommandCount = ( content.match( /\bwp\s/g ) || [] ).length;
+	const containsSingleWPCommand = wpCommandCount === 1;
+	const containsAngleBrackets = /<.*>/.test( content );
```

Comment:
I noticed when testing that AI might respond with different placeholder characters ([reference](https://github.com/Automattic/studio/pull/232#discussion_r1654495584)). I've created [a PR](https://github.com/Automattic/studio/pull/309) to improve this but I wanted to share that I created it branching off from this PR.

Reply:
@fluiddot would you prefer merging this as-is or first merge https://github.com/Automattic/studio/pull/309?

Reply:
I don't have a strong preference. Since this PR is already approved, feel free to merge as-is.